### DISCUSSION TITLE
Cannot connect using mysql2:// protocol

### DISCUSSION THREAD

**Discussion by heymartinadams** [planetscale/discussion#31] [Bug Reports] [answered]
We’re using [Prisma](https://github.com/prisma) to connect to the MySQL database and Prisma throws an error if the path does not have the `mysql://` protocol.

Our path shows up as `mysql2://root@...:3306/testdb`

**Comment by heymartinadams** [comment, marked as answer]
Ah, seeing that the CLI provides a workaround with **--execute-protocol 'mysql'**

`pscale connect testdb branch --execute-protocol 'mysql' --execute '...'`.

**Comment by bitsmyth** [comment]
I am also testing PlanetScale with Prisma in a fresh environment. How did you get the migration/introspection to work? @heymartinadams 

`pscale connect test main --execute-protocol 'mysql' --execute 'npx prisma introspect'`

```
Introspecting based on datasource defined in prisma/schema.prisma …
Error: Error in connector: Error querying the database: Error querying the database: Error querying the database: Server error: `ERROR HY000 (1105): unknown database 'test''
```

```
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

OS: MacOSX

**Reply by heymartinadams** [reply]
Haven’t been able to get it to work, @NikoMontana — for many reasons. Been in touch with Prisma’s @janpio and he thinks it’ll be a while before we can get PlanetScale to play nicely with Prisma. So for now I’m remaining with AWS until they can get it to work.

**Reply by janpio** [reply]
@NikoMontana Did you check the `Include demo dataset` checkbox when creating that database? That causes the UI from Planetsacle to tell you the db is called `test` but in reality it is called `onboarding-demo`. Which is a _bit_ confusing.

If that is the case, best create a new database in Planetscale UI and use that one. All other workarounds are just more confusing.
If not, optimally open a new issue/discussion here and tag me and I will take a look.

I am currently working on a first "Prisma + Planetscale" guide that will hopefully explain most of these things.

**Reply by bitsmyth** [reply]
Yes I saw that *issue* but even when I didn´t choose to populate the database with the `demo dataset` I was not able to connect to it.

**Reply by janpio** [reply]
Ok, that is weird. Can you please open an issue or discussion at https://github.com/prisma/prisma? We are very happy to look into that.
